extensions/ql-vscode/src/run-queries.ts

```diff
@@ -204,8 +203,9 @@ export class QueryEvaluationInfo {
         });
         if (await this.hasEvalLog()) {
           this.displayHumanReadableLogSummary(queryInfo, qs);
-          if (config.isCanary()) {
-            this.parseJsonLogSummary(qs.cliServer);
+          if (config.isCanary()) { // Generate JSON summary for viewer.
+            void qs.cliServer.generateJsonLogSummary(this.evalLogPath, this.jsonEvalLogSummaryPath);
```

Comment:
How come we don't need to `await` this?

Reply:
Hm — I was thinking that because the viewer doesn't need to populate its data until the command is clicked, it might be fine to not `await` it for now (while the query is run) to help the query finish execution more quickly.

On the other hand, I can see that it may take a really long time to generate the log summary and then the user would repeatedly hit the error message that the data is not ready. I'll change it to `await` as this is under canary mode anyway. We can see if we get reports that the double summary generation is taking too long.